### ISSUE TITLE
Model output changes randomly

### ISSUE THREAD

**Issue by swertz** [HazyResearch/hyena-dna#2]
First of all, thanks for making this great tool available!

I was trying to obtain embeddings for some sequences, and noticed that the model output changed when calling it repeatedly with the same input:

```python
tokenizer = CharacterTokenizer(
        characters=['A', 'C', 'G', 'T', 'N'],
        model_max_length=32768 + 2,
        add_special_tokens=False,
        padding_side='left',
    )

model = HyenaDNAPreTrainedModel.from_pretrained(
            './checkpoints',
            'hyenadna-small-32k-seqlen',
            download=True,
            config=None,
            device="cpu",
            use_head=False,
            n_classes=2,
        )

sequence = "ATCG"

model(torch.LongTensor(tokenizer(sequence)["input_ids"]).unsqueeze(0))[0,0,:5]
# [-0.6169,  1.0377,  0.0526, -1.0487,  0.9169]
model(torch.LongTensor(tokenizer(sequence)["input_ids"]).unsqueeze(0))[0,0,:5]
# [-0.5084,  0.6375,  0.4707, -0.8912,  1.1417]
model(torch.LongTensor(tokenizer(sequence)["input_ids"]).unsqueeze(0))[0,0,:5]
# [-0.5762,  0.3669,  0.1919, -0.7438,  1.0702]
```

I would have expected the model to be deterministic... what am I missing here?

**Comment by exnx** [comment]
You’re welcome :) thanks for finding this! We forgot to put the model in eval mode.

model.eval() after instantiating should fix this.

Eric

**Comment by vincentmin** [comment]
Hi @exnx , thank you for the swift reply. I encountered this same issue in the collab notebook. In fact the `inference_single` function contains a commented out `model.eval()` that would have prevented the issue. I recommend uncommenting that line in the notebook.
Additionally, I see that the file https://github.com/HazyResearch/hyena-dna/blob/main/huggingface.py has no `model.eval()`. I haven't checked myself if the same issue occurs, but it seems likely.
Note that `with torch.inference_mode()` is not sufficient by itself, one needs to also call `model.eval()`.

**Comment by exnx** [comment]
Oh you're right, my mistake!  I'll make those changes :)

**Comment by swertz** [comment]
Thanks for the quick reply @exnx !
